### DISCUSSION TITLE
Should Process output values that aren't correctly defined

### DISCUSSION THREAD

**Discussion by jonmaddock** [ukaea/PROCESS#2917] [General]
In some cases, Process outputs values when those values are still their initialised defaults, incorrect or misleading. A good case is https://git.ccfe.ac.uk/process/process/-/merge_requests/444, where the lack of various cryogenic components causes a negative efficiency to be output. In this case, should the output be selective, i.e. only output values that are meaningful for that particular input file?

This would mean that not all input files contain all variables. Would it make more sense to include all possible outputs, but make clear which are undefined/`None`?

**Comment by mkovari** [comment]
There are already many places in PROCESS where only relevant parameters are written to the output files.  If you wanted every output file to include all possible variables, you would have to recode all of these sections.

Certainly I don't recommend doing this for the OUT.DAT file, as it would make it even longer and very difficult to read.

1. The short term fix is to include `if` statements in the output code to ensure that only relevant outputs are written.  

2. It is possible, if you really want, to recode PROCESS as follows.
- Every possible output is written to every single MFILE, with "Not applicable" against those that are not relevant
- Only the relevant outputs are included in the OUT.DAT.

3. You can group the outputs in blocks (and indeed the OUT.DAT already has headings).  
- The heading of every block appears in the MFILE, but if it is not relevant it is empty or contains only the words "Not applicable".  
-  Only the relevant outputs are included in the OUT.DAT.
